CONTRIBUTING.md

```diff
@@ -33,50 +35,57 @@ If you're starting work on an issue ticket, please leave a comment saying so. If
-    * `model/migrations` contains the database migrations for the project. Please follow the date-based naming convention of the existing files. Migrations help us update the database structure itself to support changes and new features.
-    * `model/query` contains database queries for storing and retrieving `Instance`s. They are composable, and many of them rely on the basic operations available in `simply.js`, which defines generic get/set operations, and `all.js`, which allows multiple queries to be assembled together. All database queries should be put in these files. For each query, the first order arguments are whatever the query needs to work and are how the query will eventually be called, while dependencies are declared and injected as the second order arguments.
-    * `model/trait` defines some common functionality used across many `Instance` definitions. Like the instance methods directly declared on the `Instance`s, they do simple data operations like retrieve particular properties in particular cases.
+    * `model/instance/` defines the actual `Instance`s of the various Things that get passed around, including the static and instance methods available on them. The data associated with each `Instance` (`id`, `name`, `createdAt`, etc) are direct properties on the `Instance`. There are some common methods that are called by the framework, like `forApi` which reformats and filters the data for public consumption, and `forCreate` which decorates and formats the data for storage in the database. All dependencies are declared and injected at the very top of the file, on the line that begins `module.exports = Instance(…`.
+    * `model/migrations/` contains the database migrations for the project. Please follow the date-based naming convention of the existing files. Migrations help us update the database structure itself to support changes and new features.
+    * `model/query/` contains database queries for storing and retrieving `Instance`s. They are composable, and many of them rely on the basic operations available in `simply.js`, which defines generic get/set operations, and `all.js`, which allows multiple queries to be assembled together. All database queries should be put in these files. For each query, the first order arguments are whatever the query needs to work and are how the query will eventually be called, while dependencies are declared and injected as the second order arguments.
```

Comment:
My hunch is that currying is not familiar to all JS devs. Might be good to have a link to a reading either here or perhaps in a "concepts to be familiar with" section. Ideally it would be framed as something like "if you're not sure what two+ arrow functions means, read this". For "first order arguments" and "second order arguments" here, it might help to have an example (e.g. in `create: (table, record) => ({ db }) =>`, the first order arguments are `table` and `record`, etc.)

Reply:
Technically, we don't curry at all; these are higher order functions. But I can find a place to put it once I find an article about them that isn't confusing or wrong.